docs/bundler.md

```diff
@@ -0,0 +1,174 @@
+specified in that file.
+
+## Basic usage
+
+Run the following command in your terminal to pre-fetch your project's
+dependencies. The command will download all dependencies specified in the
+**Gemfile.lock** to the specified output directory.
+
+```bash
+cd path-to-your-ruby-project
+cachi2 fetch-deps bundler
+```
+
+In addition, it will prepare the necessary environment variables and
+configuration files for the build phase. See the following sections for more
+information.
+
+### Gems
+
+Each gem has a name, version, and platform. If the platform is "ruby", it means
+that it should work on any platform Ruby runs on. Using the ruby platform means
+ignoring the current machine's platform and installing only ruby platform gems.
+As a result, gems with native extensions will be compiled from the source.
+
+However, occasionally some gems do not have a version for the ruby platform and
+are only available as pre-compiled binaries. In this case, you may need to enable
+the pre-fetching of gems for specific platforms to avoid potential failures with
+the `allow_binary` option set to `true` when running the `fetch-deps` command.
+
+```bash
+cd path-to-your-ruby-project
+cachi2 fetch-deps '{"type": "bundler", "allow_binary": "true"}'
+```
+
+By default, the `allow_binary` option is disabled.
+
+## Configuration
+
+[Bundler](https://bundler.io/v2.5/man/bundle-config.1.html#DESCRIPTION) uses
+an unorthodox system when dealing with configuration options. The highest
+precedence is given to the config file, and then to the environment variables.
+This is a current limitation of Bundler, that we had to work around. We may
+drop the workaround if this ends up being addressed in future Bundler releases.
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+We set the following configuration options to ensure that the build process
+works correctly:
+
+- BUNDLE_CACHE_PATH: "${output_dir}/deps/bundler"
+- BUNDLE_DEPLOYMENT: "true"
+- BUNDLE_NO_PRUNE: "true"
+- BUNDLE_VERSION: "system"
+
+Here is a brief description of each configuration option taken from the
+[Bundler documentation](https://bundler.io/v2.5/man/bundle-config.1.html#LIST-OF-AVAILABLE-KEYS):
+
+### BUNDLE_CACHE_PATH
+
+The directory that Bundler will place cached gems in when running `bundle package`,
+and that Bundler will look in when installing gems. Defaults to `vendor/cache`.
+
+### BUNDLE_DEPLOYMENT
+
+Disallow changes to the **Gemfile**. When the **Gemfile** is changed and the lockfile
+has not been updated, running Bundler commands will be blocked.
+
+### BUNDLE_NO_PRUNE
+
+Whether Bundler should leave outdated gems unpruned when caching.
+
+### BUNDLE_VERSION
+
+The version of Bundler to use when running under the Bundler environment. Defaults
```

Comment:
I think we should replace evrything after the first sentence with: "cachi2 sets version to 'system' that is whichever version is installed on a system that will be building the package." A verbatim quote is pretty confusing.

Reply:
Agreed, at one point I after having read the paragraph I thought we're encouraging the user to reset the variable.

Reply:
fixed description of all environment variables